packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -233,16 +238,17 @@ const GridFilterForm = React.forwardRef<HTMLDivElement, GridFilterFormProps>(
     const baseFormControlProps = rootProps.slotProps?.baseFormControl || {};
 
     const baseSelectProps = rootProps.slotProps?.baseSelect || {};
-    const isBaseSelectNative = baseSelectProps.native ?? true;
+    const isBaseSelectNative = baseSelectProps.native ?? false;
 
     const baseInputLabelProps = rootProps.slotProps?.baseInputLabel || {};
     const baseSelectOptionProps = rootProps.slotProps?.baseSelectOption || {};
 
     const { InputComponentProps, ...valueInputPropsOther } = valueInputProps;
 
-    const filteredColumns = React.useMemo(() => {
+    const { filteredColumns, selectedField } = React.useMemo(() => {
```

Comment:
There was a hidden bug that wasn't reported when we used the `native` select.
Non-native Select warns if the value is not present among `options`.

We filter the columns here, and we have to make sure to clear the `value` if it's not included in filtered columns.